### PR TITLE
[New Rule] Microsoft 365 Teams External Access Enabled

### LINKED ISSUE #572: [New Rule] O365 Teams External Access Enabled
## Description
Identifies when external access is enabled in Microsoft Teams. External access lets your Teams and Skype for Business users communicate with other users that are outside of your organization. An adversary may enable external access or add an allowed domain to exfiltrate data or maintain persistence in an environment. 

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`o365`

- **Target ECS Version:** `1.0.6`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**

## Example Logic
```
event.dataset:o365.audit and event.provider:(SkypeForBusiness or MicrosoftTeams) and event.category:web and event.action:"Set-CsTenantFederationConfiguration" and o365.audit.Parameters.AllowFederatedUsers:True and event.outcome:success
```

## Screenshot
image

### PR DESCRIPTION
**Pull request by bm11100**
## Issues
resolves #572 

## Summary
Identifies when external access is enabled in Microsoft Teams. External access lets Teams and Skype for Business users communicate with other users that are outside of your organization. An adversary may enable external access or add an allowed domain to exfiltrate data or maintain persistence in an environment.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by bm11100** [comment]
when #570 merges, the non-ecs-schema modification may no longer be needed. will confirm.

**Review by Samirbous — APPROVED** [review]

**Comment by bm11100** [comment]
I still needed the `non-ecs-schema` even after #570 merged @brokensound77

File: rules/microsoft-365/persistence_microsoft_365_teams_external_access_enabled.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2020/11/30"
+maturity = "production"
+updated_date = "2020/11/30"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies when external access is enabled in Microsoft Teams. External access lets Teams and Skype for Business users
+communicate with other users that are outside of your organization. An adversary may enable external access or add an
```
Comment (dstepanic17) [inline rules/microsoft-365/persistence_microsoft_365_teams_external_access_enabled.toml:10]:
```suggestion
communicate with other users that are outside their organization. An adversary may enable external access or add an
```

Comment (bm11100) [reply rules/microsoft-365/persistence_microsoft_365_teams_external_access_enabled.toml]: 👍

**Review by dstepanic17 — APPROVED** [review]
LGTM, left minor comment

File: etc/non-ecs-schema.json [hunk 2]
```diff
@@ -9,5 +9,8 @@
   },
   "winlogbeat-*": {
     "winlog.event_data.OriginalFileName": "keyword"
+  },
+  "filebeat-*": {
+    "o365.audit.Parameters.AllowFederatedUsers": "keyword"
```
Comment (brokensound77) [inline etc/non-ecs-schema.json:14]: Dependent on the resolution from https://github.com/elastic/detection-rules/pull/578#issuecomment-740132064

Anything under `Parameters` or any other `X.*` of type object

If no resolution by COB, we should just merge this with the exception and resolve in the feature branch

**Review by brokensound77 — APPROVED** [review]
Left a comment on the bad field, otherwise LGTM 👍

**Pull request merged**